homeassistant/components/fortios/device_tracker.py

```diff
@@ -106,17 +112,12 @@ def get_device_name(self, device):
             return None
 
         for client in data["results"]:
-            if client["mac"] == device:
-                try:
-                    name = client["hostname"]
-                    _LOGGER.debug("Getting device name=%s", name)
+            if "master_mac" in client:
+                if client["master_mac"] == device:
+                    name = ""
+                    if "hostname" in client:
+                        name = client["hostname"]
+                    else:
+                        name = client["master_mac"].replace(":", "_")
                     return name
```

Comment:
```suggestion
            if "master_mac" in client and client["master_mac"] == device:
                if "hostname" in client:
                    name = client["hostname"]
                else:
                    name = client["master_mac"].replace(":", "_")
                return name
```

Reply:
This comment wasn't addressed.

Reply:
sorry - done.

Reply:
your proposed changes has been tested. I just forgot to resolve.